airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -100,25 +100,67 @@ def get_dags(
 @provide_session
 def patch_dag(*, dag_id: str, update_mask: UpdateMask = None, session: Session = NEW_SESSION) -> APIResponse:
     """Update the specific DAG"""
+    try:
+        patch_body = dag_schema.load(request.json, session=session)
+    except ValidationError as err:
+        raise BadRequest(detail=str(err.messages))
+    if update_mask:
+        patch_body_ = {}
+        if update_mask != ['is_paused']:
+            raise BadRequest(detail="Only `is_paused` field can be updated through the REST API")
+        patch_body_[update_mask[0]] = patch_body[update_mask[0]]
+        patch_body = patch_body_
     dag = session.query(DagModel).filter(DagModel.dag_id == dag_id).one_or_none()
     if not dag:
         raise NotFound(f"Dag with id: '{dag_id}' not found")
+    dag.is_paused = patch_body['is_paused']
+    session.flush()
+    return dag_schema.dump(dag)
+
+
+@security.requires_access([(permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG)])
+@format_parameters({'limit': check_limit})
+@provide_session
+def patch_dags(limit, session, offset=0, only_active=True, tags=None, dag_id_pattern=None, update_mask=None):
+    """Patch multiple DAGs."""
     try:
         patch_body = dag_schema.load(request.json, session=session)
     except ValidationError as err:
-        raise BadRequest("Invalid Dag schema", detail=str(err.messages))
+        raise BadRequest(detail=str(err.messages))
     if update_mask:
         patch_body_ = {}
-        if len(update_mask) > 1:
+        if update_mask != ['is_paused']:
             raise BadRequest(detail="Only `is_paused` field can be updated through the REST API")
         update_mask = update_mask[0]
-        if update_mask != 'is_paused':
-            raise BadRequest(detail="Only `is_paused` field can be updated through the REST API")
         patch_body_[update_mask] = patch_body[update_mask]
         patch_body = patch_body_
-    setattr(dag, 'is_paused', patch_body['is_paused'])
-    session.commit()
-    return dag_schema.dump(dag)
+    if only_active:
+        dags_query = session.query(DagModel).filter(~DagModel.is_subdag, DagModel.is_active)
+    else:
+        dags_query = session.query(DagModel).filter(~DagModel.is_subdag)
+
+    if dag_id_pattern == '~':
+        dag_id_pattern = '%'
```

Comment:
What happens when the dag_id_pattern is None? Will it update all DAGs or it won't? 
I'll still do a real world testing though, just curious

Reply:
The current behaviour is to update all DAGs. So yeah this new logic seems wrong in that case, good catch.

Reply:
`dag_id_pattern` should never be `None` since it is a required parameter. 
When I pass `None` I get the following response: `Dag with id: ''None'' not found`.